### PR TITLE
[selenium-webdriver] add ShadowRoot definition

### LINKED ISSUE #57681: ShadowRoot type
I need help with selenium-webdriver 4.1.0 to get the shadowRoot (I use Typescript as the main language in my project; I use @types/selenium-webdriver 4.0.16).

```javascript 
public async getExtShadowRoot(selectorShadowHost: string): Promise<unknown> {
    let shadowHost: WebElementPromise;
    await (shadowHost = this.driver.findElement(By.css(selectorShadowHost)));
    return this.driver.executeScript("return arguments[0].shadowRoot", shadowHost);
  }
```
From the above method call I get the following Promise:

```json
Promise {
  ShadowRoot {
    driver_: thenableWebDriverProxy {
      session_: [Promise],
      executor_: [Executor],
      fileDetector_: null,
      onQuit_: [Function: onQuit],
      then: [Function: bound then],
      catch: [Function: bound catch]
    },
    id_: '58170f1b-0247-40b4-8916-81d45f8642ed'
  }
}
```
@types/selenium-webdriver 4.0.16 does not export the ShadowRoot type.
What I am looking for is to obtain the elements contained in ShadowRoot to be able to interact with them through Selenium. Thanks.

### PR DESCRIPTION
**Pull request by ajdor**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/SeleniumHQ/selenium/commit/4e24e999b7f12510793e8bf515314bf2fa7ae5e8
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

This PR adds the ShadowDOM/ShadowRoot definitions part of the selenium-webdriver 4.1.0 update.
Closes https://github.com/DefinitelyTyped/DefinitelyTyped/issues/57681

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## selenium-webdriver ([<kbd>unpkg</kbd>](https://unpkg.com/browse/selenium-webdriver@latest/))
was missing the following properties:
1. RelativeBy
1. withTagName
1. locateWith
1. ThenableWebDriver
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against dd0509cf9430522ae15202555792c4b1194b6158
</p>

File: types/selenium-webdriver/lib/webdriver.d.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+/**
+ * ShadowRootPromise is a promise that will be fulfilled with a WebElement.
+ * This serves as a forward proxy on ShadowRoot, allowing calls to be
+ * scheduled without directly on this instance before the underlying
+ * ShadowRoot has been fulfilled.
+ *
+ * @implements { IThenable<!ShadowRoot>}
+ * @final
+ */
+export class ShadowRootPromise extends Promise<ShadowRoot> {
```
Comment (SupernaviX) [inline types/selenium-webdriver/lib/webdriver.d.ts:101]: It looks like`ShadowRootPromise` is supposed to work as both a `Promise<ShadowRoot>` and a `ShadowRoot`.

I'd recommend using the same interface merging trick that we used for that in [WebElementPromise](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/dd0509cf9430522ae15202555792c4b1194b6158/types/selenium-webdriver/index.d.ts#L2692).

```suggestion
export interface ShadowRootPromise extends Promise<ShadowRoot> {}

export class ShadowRootPromise extends ShadowRoot {
```

**Review by SupernaviX — COMMENTED** [review]
Looks good for the most part! But I think the ShadowRootPromise typings are a bit off

File: types/selenium-webdriver/lib/webdriver.d.ts (hunk 1 above)
Comment (ajdor) [reply types/selenium-webdriver/lib/webdriver.d.ts]: Done, thanks !

**Review by SupernaviX — APPROVED** [review]

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by ajdor** [comment]
Ready to merge

**Pull request merged**